include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -206,6 +206,23 @@ struct is_passed_directly<Iter, ::std::enable_if_t<Iter::is_passed_directly::val
 {
 };
 
+//support std::vector::iterator with usm host / shared allocator as passed directly
+template <class Iter>
+struct is_passed_directly<
+    Iter,
+    ::std::enable_if_t<
+        ::std::is_same_v<
+            Iter, typename ::std::vector<typename ::std::iterator_traits<Iter>::value_type,
+                                         typename sycl::usm_allocator<typename ::std::iterator_traits<Iter>::value_type,
+                                                                      sycl::usm::alloc::shared>>::iterator> ||
+        ::std::is_same_v<
+            Iter, typename ::std::vector<typename ::std::iterator_traits<Iter>::value_type,
+                                         typename sycl::usm_allocator<typename ::std::iterator_traits<Iter>::value_type,
+                                                                      sycl::usm::alloc::host>>::iterator>>>
```

Comment:
Probably, should be a `sycl::usm::alloc::device` ?

Reply:
This doesn't exist. sycl doesnt define a device usm_allocator.